utilities/transactions/write_prepared_txn.cc

```diff
@@ -63,6 +62,76 @@ Iterator* WritePreparedTxn::GetIterator(const ReadOptions& options,
   return write_batch_.NewIteratorWithBase(column_family, db_iter);
 }
 
+// TODO(myabandeh): deprected
+struct CFKeyComparator {
+  bool operator()(const std::pair<uint32_t, Slice>& lhs,
+                  const std::pair<uint32_t, Slice>& rhs) const {
+    if (lhs.first != rhs.first) {
+      return lhs.first < rhs.first;
+    }
+    return lhs.second.compare(rhs.second) < 0;
+  }
+};
+struct SetComparator {
+  SetComparator() : user_comparator_(BytewiseComparator()) {}
+  SetComparator(const Comparator* user_comparator)
+      : user_comparator_(user_comparator ? user_comparator
+                                         : BytewiseComparator()) {}
+  bool operator()(const Slice& lhs, const Slice& rhs) const {
+    return user_comparator_->Compare(lhs, rhs) < 0;
+  }
+
+ private:
+  const Comparator* user_comparator_;
+};
+// Count the number of sub-batches inside a batch. A sub-batch does not have
+// duplicate keys.
+struct SubBatchCounter : public WriteBatch::Handler {
+  SubBatchCounter(std::map<uint32_t, const Comparator*>& comparators)
+      : comparators_(comparators), batches_(1) {}
+  std::map<uint32_t, const Comparator*>& comparators_;
```

Comment:
you can still use unordered_map.

Reply:
given that there will be a few cf in each txn (perhaps only 1), and given that cf is integer i think binary search tree would be as good as hash map if not faster.

Reply:
ok. yeah it is really hard to tell which one is better when the size is really small.